### DISCUSSION TITLE
How should I modify the color of a particular text in the text

### DISCUSSION THREAD

**Discussion by mcclove** [pixijs/pixijs#9398] [Ideas]
pixi7.2.4。
let renderer = PIXI.autoDetectRenderer(800, 600);
        let stage = new PIXI.Container();

        let text = new PIXI.Text('Hello PIXI!', {
            fill: 0xFFFFFF,   // 白色
            fontSize: 64
        });
        text.style.fill[2] = 0xFFFF00;   // 设置第3个字符为黄色

        stage.addChild(text);
        document.getElementById("pixi-container").appendChild(renderer.view);
        renderer.render(stage);   // 渲染
        
        Why is this line text. style. fill [2]=0xFFFF00 not executed

**Comment by SuperSodaSea** [comment]
`Text` don't support specifying colors for specific characters. You can use `HTMLText` instead (supported in PixiJS 7.2.0+):

```js
const text = new PIXI.HTMLText(
    'Hello, <span style="color: #F10061">PixiJS</span>!',
    { fontSize: 32, fill: 0xFFFFFF });
app.stage.addChild(text);
```

Demo: <https://pixiplayground.com/#/edit/xm3ONb8ibfvpZU2tH5rRt>

**Reply by mcclove** [reply]
Wow, I tried. It's okay, thank you. It seems that I still need to continue learning
